crates/flashnet/src/api.rs

```diff
@@ -0,0 +1,417 @@
+        self.get_request("v1/ping", None::<()>).await
+    }
+
+    async fn ensure_feature_status(&self, feature_name: FeatureName) -> Result<(), FlashnetError> {
+        let feature_status_cache = self
+            .cache_store
+            .get::<Vec<FeatureStatus>>(FEATURE_STATUSES_CACHE_KEY)
+            .await?;
+        let feature_statuses = if let Some(feature_statuses) = feature_status_cache {
+            feature_statuses
+        } else {
+            let feature_statuses = self.config_feature_status().await?;
+            self.cache_store
+                .set(
+                    FEATURE_STATUSES_CACHE_KEY,
+                    &feature_statuses,
+                    FEATURE_STATUSES_TTL_MS.into(),
+                )
+                .await?;
+            feature_statuses
+        };
+        let feature_status = feature_statuses
+            .into_iter()
+            .find(|fs| fs.feature_name == feature_name);
+        match feature_status {
+            Some(fs) => {
+                if fs.enabled {
+                    Ok(())
+                } else {
+                    Err(FlashnetError::Generic(format!(
+                        "Feature {:?} is disabled: {}",
+                        feature_name,
+                        fs.reason
+                            .unwrap_or_else(|| "No reason provided".to_string())
+                    )))
+                }
+            }
+            None => Err(FlashnetError::Generic(format!(
+                "Feature {feature_name:?} not found in status list"
+            ))),
+        }
+    }
+
+    async fn ensure_min_amounts(
+        &self,
+        asset_in_address: &str,
+        asset_out_address: &str,
+        amount_in: u128,
+        min_amount_out: Option<u128>,
+    ) -> Result<(), FlashnetError> {
+        let min_amounts_cache = self
+            .cache_store
+            .get::<Vec<MinAmount>>(MIN_AMOUNTS_CACHE_KEY)
+            .await?;
+        let min_amounts = if let Some(min_amounts) = min_amounts_cache {
+            min_amounts
+        } else {
+            let min_amounts = self.config_min_amounts().await?;
+            self.cache_store
+                .set(
+                    MIN_AMOUNTS_CACHE_KEY,
+                    &min_amounts,
+                    MIN_AMOUNTS_TTL_MS.into(),
+                )
+                .await?;
+            min_amounts
+        };
+        let min_amounts_map = min_amounts
+            .into_iter()
+            .filter(|ma| ma.enabled)
+            .map(|ma| (ma.asset_identifier, ma.min_amount))
+            .collect::<HashMap<_, _>>();
+        if let Some(&min_in) = min_amounts_map.get(asset_in_address) {
+            if amount_in < min_in {
+                return Err(FlashnetError::Generic(format!(
+                    "Amount in {amount_in} is less than minimum required {min_in} for asset {asset_in_address}",
+                )));
+            }
+            return Ok(());
```

Comment:
We skip here without checking the min_amount_out ?

Reply:
To me it looks like the min_amount_out is not validated in this case. It looks like a bug unless it is not used when min_in is set.